### PR TITLE
Version selection complete, updating Course Offering test data

### PR DESCRIPTION
**Pull request by hannahbergam**
The logic to display the version selector is very complex. After looking into what it might entail to rewrite, I decided to reuse the existing component (no need to reinvent the wheel). Unfortunately, it requires our course offerings to include the more verbose course_versions (associated with the assignment dropdowns we currently use). 

I switched out our course version model call to go back to using the existing call and deleted the one we built custom for quick assign. Then I updated all the test data and the tests to match (It involves more data than I'm using currently, but I imagine a robust testing future will also include a storybook, where all these elements will come into play). 

Now, the unit version dropdown works as-is. As soon as a user selects an offering, if there is only one version or there is a recommended version, that versionId is automatically selected. The user can then view the dropdown to see the recommended and lang offerings, and select a different one if desired. 

Screenshot 2023-03-17 at 2 56 47 PM

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/TEACH-265
- https://codedotorg.atlassian.net/browse/TEACH-327

## Follow-up work

The unit dropdown work will come next. Unfortunately, that was not written in a small reusable component, so will need to be copied over and converted to work here. 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx [hunk 1]
```diff
@@ -31,24 +37,50 @@ export function renderRows(courseData, sectionCourse, updateCourse) {
...
       </label>
     </div>
   ));
 }
+
+function updateSectionCourse(updateCourse, course) {
+  let courseVersionId;
+  const courseVersions = {};
+  course.course_versions.map(cv => {
+    courseVersions[cv[1].id] = cv[1];
```
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:74]: why `cv[1]`?

File: apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      versionObject[cv[1].id] = cv[1];
+    });
+    return versionObject;
+  };
+
+  return (
+    <div className={moduleStyles.buttonRow}>
+      <div className={moduleStyles.buttonsInRow}>
+        {courseOffering && (
+          <AssignmentVersionSelector
```
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx:34]: yay for being able to reuse this!

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:74]: The data structure that exists is super weird- it is shaped like this (see photo) where the id of the course version is index zero in an array, and all its info is index 1
Screenshot 2023-03-21 at 9 22 49 AM

File: apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx (hunk 2 above)
Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx:34]: 🎉

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:74]: oof that's annoying but that does vaguely ring some bells. Can you add a comment explaining that? Maybe point to the test data file or something?

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx [hunk 3]
```diff
@@ -31,24 +37,50 @@ export function renderRows(courseData, sectionCourse, updateCourse) {
...
+  });
+  if (Object.keys(courseVersions).length === 1) {
+    courseVersionId = Object.values(courseVersions)[0].id;
+  } else {
+    courseVersionId = Object.values(courseVersions)?.find(
+      versions => versions.is_recommended
+    )?.id;
```
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:81]: This is definitely an edge case but is there any default we want to have if there aren't any recommended courses? Or do we just assume that won't happen?

File: apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            selectedCourseVersionId={sectionCourse.versionId}
+            courseVersions={prepareCourseVersions()}
+            onChangeVersion={id => updateCourseDetail(VERSION_ID, id)}
+          />
+        )}
+        {sectionCourse?.unitId && (
```
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx:40]: My brain is still pre-caffeine -- what is this section doing? I thought the unit selector was coming later?

Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx:40]: Hi- yes I'll be doing that later. I had set this up as a placeholder but it isn't actually working yet so just followed through with all the version portion... My bad. It's going to be the very next task but am happy to delete that part for now

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:74]: So annoying! And yes I'll add a comment!

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx (hunk 3 above)
Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:81]: I copied this logic over from our existing flow, so if that felt okay then I think I'm okay with that risk here too!

File: apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx (hunk 4 above)
Comment (bethanyaconnor) [reply apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx:40]: It's okay -- I don't think it's breaking anything and it's not live yet. I just haven't really reviewed it as I assume you're going to change it in the next PR.

**Review by bethanyaconnor — APPROVED** [review]

File: apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx (hunk 4 above)
Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx:40]: Thanks! And yes I haven't done the work to add the unitId in the first place, so it just isn't popping into this logic yet (but it's at least a start)

**Pull request merged**